src/ResourceManager/Common/Commands.ResourceManager.Common/ArgumentCompleters/ResourceTypeCompleter.cs

```diff
@@ -0,0 +1,149 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters
+{
+    using Commands.Common.Authentication;
+    using Commands.Common.Authentication.Abstractions;
+    using Management.Internal.Resources;
+    using Management.Internal.Resources.Models;
+    using Properties;
+    using Rest.Azure;
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Management.Automation;
+
+    /// <summary>
+    /// This attribute will allow the user to autocomplete the -ResourceType parameter of a cmdlet with valid resource types
```

Comment:
Can we split this into a separate PR?  I want to make sure we get in as much of this as possible